src/bindings/python/src/pyopenvino/graph/shape.cpp

```diff
@@ -44,9 +44,30 @@ void regclass_graph_Shape(py::module m) {
         if (key < 0) {
             key += v.size();
         }
-        return v[key];
+        if (key >= 0 && key < static_cast<int64_t>(v.size())) {
+            return v[key];
+        }
+        throw py::index_error("Index out of range");
+    });
+
+    auto compareShape = [](const ov::Shape& self, const auto& container) {
```

Comment:
```cpp
    auto compareShape = [](const ov::Shape& self, const auto& container) {
```
This will not work with cpp11 (no generic lambdas),  use template function instead.  Use `snake_case` for functions/variables names.

Consider change this comparator to:
```cpp
return self.size() == container.size() && std::equal(self.begin(), self.end(), container.begin(), bin_predicate);
```